cmd/upgrade/upgrade.go

```diff
@@ -0,0 +1,112 @@
+package upgrade
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/rancher/wins/cmd/upgrade/internal/powershell"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var _upgradeFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "wins-args",
+		Usage: "[optional] Arguments to pass onto wins srv app run --register",
+	},
+	cli.BoolFlag{
+		Name:  "debug",
+		Usage: "[optional] whether to print debugging logs from performing the upgrade",
+	},
+}
+
+const upgradePS1Fmt = `& {
+	$ErrorActionPreference = "Stop";
+
+	$newBinPath = "%[1]s"
+	$winsArgs = "%[2]s"
+
+	$winsCmd = (Get-CimInstance Win32_Service -Filter 'Name = "rancher-wins"')
+
+	$winsSvc = Get-Service -Name rancher-wins
+	if ($winsSvc) {
+		$winsSvc | Stop-Service;
+	}
+	$winsPrc = Get-Process -Name wins -ErrorAction SilentlyContinue
+	if ($winsPrc) {
+		$winsPrc | Stop-Process -Force
+	}
+
+	if ($winsCmd -ne $null) {
+		$winsCmdPath = $winsCmd.PathName
+		$currBinPath = $winsCmdPath.Split(' ')[0]
+		if ($currBinPath -ne $newBinPath) {
+			Copy-Item -Recurse -Force -Path $newBinPath -Destination $currBinPath | Out-Null
+		}
+	} else {
+		$currBinPath = $newBinPath
+	}
+
+	Invoke-Expression "& $currBinPath srv app run --register $winsArgs"
+	Start-Service -Name rancher-wins;
+	Write-Host "Upgraded rancher-wins"
```

Comment:
Is this whitespace going to be an issue?
EDIT: and for other places in the file

Reply:
When I tried it out in the rancher-wins-upgrader chart with a local copy, it did not cause any issues. So I think the whitespace should be fine

Reply:
Agreed, just wanted to double check